ste/folderCreationTracker.go

```diff
@@ -9,6 +9,78 @@ import (
 	"github.com/Azure/azure-storage-azcopy/v10/common"
 )
 
+type TrieNode struct {
```

Comment:
May be better to put this implementation of Trie in a seperate file and have some UT around it.